### ISSUE TITLE
add --datadir flag

### ISSUE THREAD

**Issue by bortzmeyer** [ethereum/mist#682]
```
% geth version
Geth
Version: 1.4.3-stable
Protocol Versions: [63 62 61]
Network Id: 1
Go Version: go1.5.1
OS: linux
GOPATH=/usr/lib/go/src/pkg
GOROOT=/usr/lib/go
```

Ubuntu 32 buts: Linux tyrion 4.2.0-30-generic #36-Ubuntu SMP Fri Feb 26 00:57:19 UTC 2016 i686 i686 i686 GNU/Linux

Testnet

Mist version: 0.7.4

```
% ./Ethereum-Wallet 
[2016-05-22 10:59:16.800] [INFO] main - Starting in Wallet mode
[2016-05-22 10:59:23.863] [WARN] nodeConnector - NODECONNECTOR ERROR { [Error: connect ENOENT /home/stephane/.ethereum/geth.ipc]
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'connect',
  address: '/home/stephane/.ethereum/geth.ipc' }
Error: connect ENOENT /home/stephane/.ethereum/geth.ipc
    at Object.exports._errnoException (util.js:856:11)
    at exports._exceptionWithHostPort (util.js:879:20)
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
[2016-05-22 10:59:24.861] [INFO] ipcCommunicator - Backend language set to:  en-US
[2016-05-22 10:59:25.053] [WARN] nodeConnector - NODECONNECTOR ERROR { [Error: connect ENOENT /home/stephane/.ethereum/geth.ipc]
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'connect',
  address: '/home/stephane/.ethereum/geth.ipc' }
Error: connect ENOENT /home/stephane/.ethereum/geth.ipc
    at Object.exports._errnoException (util.js:856:11)
    at exports._exceptionWithHostPort (util.js:879:20)
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
[2016-05-22 10:59:26.153] [INFO] main - Node type:  geth
[2016-05-22 10:59:26.153] [INFO] main - Network:  main
[2016-05-22 10:59:26.154] [INFO] ethereumNodes - Start node from /home/stephane/tmp/Ethereum-Wallet-linux32-0-7-4/resources/node/geth/geth
[2016-05-22 10:59:26.161] [INFO] ethereumNodes - Stopping nodes...
[2016-05-22 10:59:26.161] [INFO] ethereumNodes - Starting geth node...

```

And then of course Wallet fails since there *is* an existing geth:

```
Node type: geth
Network: main
Platform: linux (Architecure ia32)

...ta
I0522 11:04:16.283547 ethdb/database.go:169] closed db:/home/stephane/.ethereum/chaindata
I0522 11:04:16.288227 cmd/utils/flags.go:601] WARNING: No etherbase set and no accounts found as default
I0522 11:04:16.289287 ethdb/database.go:82] Alloted 512MB cache and 1024 file handles to /home/stephane/.ethereum/chaindata
I0522 11:04:16.705945 ethdb/database.go:82] Alloted 16MB cache and 16 file handles to /home/stephane/.ethereum/dapp
I0522 11:04:16.722459 eth/backend.go:170] Protocol Versions: [63 62 61], Network Id: 1
I0522 11:04:16.722706 eth/backend.go:199] Blockchain DB Version: 3
I0522 11:04:16.723738 core/blockchain.go:206] Last header: #0 [d4e56740…] TD=17179869184
I0522 11:04:16.723783 core/blockchain.go:207] Last block: #0 [d4e56740…] TD=17179869184
I0522 11:04:16.723806 core/blockchain.go:208] Fast block: #0 [d4e56740…] TD=17179869184
I0522 11:04:16.725400 p2p/server.go:311] Starting Server
Fatal: Error starting protocol stack: listen udp :30303: bind: address already in use
```

**Comment by luclu** [comment]
@bortzmeyer please try to close the external geth instance and/or reboot. (Also update your geth to version 1.4.4.).
This error can occur if e.g. you have an external geth instance specified with some custom datadir and try to open the wallet.

**Comment by bortzmeyer** [comment]
@luclu But I don't want to close my geth. I want to use my own one, with my own parameters.

**Comment by luclu** [comment]
That's fine, let's create a softlink to your know datadir with `ln -s`.

**Comment by bortzmeyer** [comment]
@luclu Your trick with the symbolic link works fine, many thanks; I can now run the Wallet.

Why no --datadir option for the wallet?

**Comment by luclu** [comment]
Let's put that on the list. Another reference https://github.com/ethereum/mist/issues/156#issuecomment-216470568

**Comment by hiddentao** [comment]
PR in #689

**Comment by luclu** [comment]
Will be available in the next version 0.7.5.
